data-prepper-pipeline-parser/src/test/resources/transformation/rules/documentdb-rule.yaml

```diff
@@ -1,3 +1,4 @@
+plugin_name: "documentdb"
```

Comment:
As this is for testing, let's add test in the name. Maybe `test-documentdb`.

Reply:
@srikanthjg do you want to make this small change?

Reply:
I think it will be inconsistent even within the same rule yaml. The apply_when condition for the rule yaml, we still define it as documentdb and it would looks as follows: 
```
plugin_name: "documentdbtest"
apply_when:
  - "$..source.documentdb"
 ```
We still want use documentdb and not documentdbtest in the apply_when condition as that is what is used in the pipeline yaml; and that cannot be mocked that easily.

The plugin name in the rule yaml is only used to get the corresponding template, but it seems inconsistent if we use documentdb vs documentdbtest even within the same file for the same plugin we want to test.

Reply:
Why wouldn't this work?

```
plugin_name: "documentdbtest"
apply_when:
  - "$..source.documentdbtest"
```

These files are only used for unit testing, right? So, there is no need to keep the names matching the others.

Reply:
This is where you use it:

https://github.com/opensearch-project/data-prepper/pull/4601/files#diff-40ef318469b1b8fb28c6ee16a998f746921ecee5471a9f481d497617b60a2180R38

Reply:
i will need to change like 20 files(test yaml files) in the test as all of them use documentdb. i cant just change the rule to documentdbtest and make it work. The userconfig, expected output and template depends on this. can i add this as another PR? it should it mostly replace all but i see some other issues with it.